polygeist/include/mlir/Dialect/Polygeist/Transforms/Passes.td

```diff
@@ -16,6 +16,40 @@ def AffineCFG : Pass<"affine-cfg"> {
   let constructor = "mlir::polygeist::replaceAffineCFGPass()";
 }
 
+def AffineReduction : Pass<"detect-reduction"> {
+  let summary = "Detect reductions in affine.for";
+  let constructor = "mlir::polygeist::detectReductionPass()";
+  let statistics = [
+    Statistic<"numReductions", "num-reductions", "Number of reductions detected">
+  ];
+}
+
+def ArgumentPromotion : Pass<"arg-promotion", "mlir::ModuleOp"> {
+  let summary = "Promote kernel arguments";
+  let description = [{
+    Peel struct members from the argument pack passed to a SYCL kernel. 
```

Comment:
not really to a SYCL kernel, it is the function called by a SYCL kernel.

Reply:
clarified